Fixed error when yaml file only contains comments

LoXeras opened the pull request:
Hello, 

We have a lot of YAML files in our environment and also some that only have comments in it (also no --- only comments).

example:
#foo: bar

These files are breaking the YAML lint process. 
I have expanded the file check mechanism that it doesn't break anymore.
I have also added a Test to it.

Tested it out on all our files and the YAML lint test goes trough.

Let me know if i can improve or change something.

Kind regards
LoXeras

LoXeras committed:
added ignore to files that only contain commets

shortdudey123 commented on lib/yamllint/linter.rb:
```diff
@@ -43,10 +43,20 @@ def check(path)
-        errors[path] = error_array unless error_array.empty?
+      commentcount = false
+      File.readlines(path).each do |line|
+        if(!(line.include? "#"))
```
Travis did not run but I think this will fail rubocop since you use if not instead of unless

LoXeras replied:
changed to unless

shortdudey123 commented on spec/linter_spec.rb:
```diff
@@ -48,6 +48,7 @@
 
   it 'should be unhappy with an invalid YAML file' do
     expect(linter.check(spec_data('invalid.yaml'))).to be(false)
+    expect(linter.check(spec_data('invalid_only_comment.yaml'))).to be(false)
```
Please separate this to a separate `it` lock

LoXeras replied:
separated it

shortdudey123 commented:
Please add a line to the unreleased section of the read me following the existing format and use the Github PR title as the description.

LoXeras committed:
Fixed error when yaml file only contains comments

LoXeras commented:
Thank yo for your feedback, i made the changes that you have written and also executed rubucop and fixed some linting.
I also added the PR to the "unreleased changes"

I had to update the Commit author as i made a mistake locally, author name is correct now. Sorry for that.

Let me know if there is anything else to change

LoXeras

shortdudey123 commented:
Used #42 instead since it is simpler and more correct.
